### DISCUSSION TITLE
Kernel I2C probing not called

### DISCUSSION THREAD

**Discussion by cyring** [cyring/CoreFreq#301] [Q&A]
I'm working on an I2C driver implementation. The aim is to query registers through SB-RMI mailbox protocol. So far it is tested with the `AMD EPYC 7281 16-Core Processor`

The work in progress is within commit https://github.com/cyring/CoreFreq/commit/071f783ae142bafd3adde0ec2b0c4c64f7991399

Currently I'm stuck with the device probing function never being called: anyone with such experience ?

**Comment by jowa2021** [comment]
/sys/bus/i2c/devices/i2c-0/subsystem/drivers/intel_soc_pmic_i2c

This is just an example pathway. This is not from same cpu.

Have similar issue, try changing kernel ??

**Reply by jowa2021** [reply]
Can you check dmesg and pci to corresponding devices on your system ??
![image](https://user-images.githubusercontent.com/80190819/143787039-0ec86aaf-277b-45bb-ba71-2a893175102e.png)

**Reply by cyring** [reply]
Thanks I will try another kernel (after new server renting)
Will let you know.

**Comment by jowa2021** [comment]
oh sorry i didnt realize you would have to do it that way, i thought maybe you could reboot or build a new

**Reply by cyring** [reply]
Indeed SB-RMI is apparently the protocol used by the BMC monitoring on EPYC servers. Those Registers are specified for Zen V1, thus Naples. I don't know if they can be addressed with Desktop & Threadripper.
It is said that SB-RMI is a SMBUS protocol through one of the height I2C, `0x78` for socket `0` and `0x70` for socket `1`

We will find Control, Status and Alert registers. The last is about a Software interrupt bit set by firmware when the queried data are available for fetching.
Two bytes buffers are used, one for each direction of the protocol.

AMD has upstreamed it's HWMON driver @ 
https://elixir.bootlin.com/linux/latest/source/drivers/hwmon/sbrmi.c

**Comment by cyring** [comment]
Dev ceased
